File: app/models/inventory_item.py
from typing import Optional, TYPE_CHECKING, List

from sqlmodel import SQLModel, Field, Relationship, UniqueConstraint

if TYPE_CHECKING:
    from .transaction import Transaction


class InventoryItem(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str
    barcode: str = Field(index=True, unique=True)
    description: Optional[str] = None
    quantity: int = 0
    sku: Optional[str] = None
    # Either an externally hosted URL or a locally uploaded file name (stored under /uploads)
    image_url: Optional[str] = None
    image_path: Optional[str] = None

    # Relationship to transactions (item.transactions)
    transactions: List["Transaction"] = Relationship(back_populates="item")

    class Config:
        arbitrary_types_allowed = True

    __table_args__ = (
        # Enforce uniqueness at the DB level for barcode
        UniqueConstraint("barcode", name="uq_inventory_item_barcode"),
    )

File: xcode/InventoryApp/InventoryApp/InventoryItem.swift
import Foundation

struct InventoryItem: Identifiable, Codable {
    let id: Int
    let barcode: String
    let name: String
    let description: String?
    var quantity: Int?
    let sku: String?
    let imageURL: String?
    let imagePath: String?
    let previewURL: String?

    enum CodingKeys: String, CodingKey {
        case id, barcode, name, description, quantity, sku
        case imageURL = "image_url"
        case imagePath = "image_path"
        case previewURL = "preview_url"
    }
}
